test/hpke_test.c

```diff
@@ -0,0 +1,1410 @@
+        goto end;
+    if (!TEST_true(OSSL_HPKE_export(rctx, rexp, 32,
+                                    (unsigned char *)estr, strlen(estr))))
+        goto end;
+    /* check a 2nd call with same input gives same output */
+    if (!TEST_true(OSSL_HPKE_export(rctx, rexp2, 32,
+                                    (unsigned char *)estr, strlen(estr))))
+        goto end;
+    if (!TEST_true(TEST_mem_eq(rexp, 32, rexp2, 32)))
+        goto end;
+    if (!TEST_true(TEST_mem_eq(exp, 32, rexp, 32)))
+        goto end;
+    OSSL_HPKE_CTX_free(ctx);
+    OSSL_HPKE_CTX_free(rctx);
+    EVP_PKEY_free(privp);
+    return 1;
+end:
+    OSSL_HPKE_CTX_free(ctx);
+    OSSL_HPKE_CTX_free(rctx);
+    EVP_PKEY_free(privp);
+    return 0;
+}
+
+/**
+ * @brief Check mapping from strings to HPKE suites
+ * @return 1 for success, other otherwise
+ */
+static int test_hpke_suite_strs(void)
+{
+    int overallresult = 1;
+    int kemind = 0;
+    int kdfind = 0;
+    int aeadind = 0;
+    int sind = 0;
+    char sstr[128];
+    OSSL_HPKE_SUITE stirred;
+
+    for (kemind = 0;
+         kemind != (sizeof(kem_str_list) / sizeof(char *));
+         kemind++) {
+        for (kdfind = 0;
+             kdfind != (sizeof(kdf_str_list) / sizeof(char *));
+             kdfind++) {
+            for (aeadind = 0;
+                 aeadind != (sizeof(aead_str_list) / sizeof(char *));
+                 aeadind++) {
+                snprintf(sstr, 128, "%s,%s,%s",
+                         kem_str_list[kemind],
+                         kdf_str_list[kdfind],
+                         aead_str_list[aeadind]);
+                if (TEST_true(OSSL_HPKE_str2suite(sstr, &stirred)) != 1) {
+                    overallresult = 0;
+                }
+            }
+        }
+    }
+    for (sind = 0;
+         sind != (sizeof(bogus_suite_strs) / sizeof(char *));
+         sind++) {
+        char dstr[128];
+
+        sprintf(dstr, "str2suite: %s", bogus_suite_strs[sind]);
+        if (TEST_false(OSSL_HPKE_str2suite(bogus_suite_strs[sind],
+                                           &stirred)) != 1) {
+            overallresult = 0;
+        }
+    }
+    /* last check on empty string */
+    if (!TEST_false(OSSL_HPKE_str2suite("", &stirred)))
+        overallresult = 0;
+
+    return overallresult;
+}
+
+/**
+ * @brief try the various GREASEy APIs
+ * @return 1 for success, other otherwise
+ */
+static int test_hpke_grease(void)
+{
+    int overallresult = 1;
+    OSSL_HPKE_SUITE g_suite;
+    unsigned char g_pub[OSSL_HPKE_MAXSIZE];
+    size_t g_pub_len = OSSL_HPKE_MAXSIZE;
+    unsigned char g_cipher[OSSL_HPKE_MAXSIZE];
+    size_t g_cipher_len = 266;
+    size_t clearlen = 128;
+    size_t expanded = 0;
+    size_t enclen = 0;
+    size_t ikmelen = 0;
+
+    memset(&g_suite, 0, sizeof(OSSL_HPKE_SUITE));
+    /* GREASEing */
+    if (TEST_true(OSSL_HPKE_get_grease_value(testctx, NULL, NULL, &g_suite,
+                                             g_pub, &g_pub_len,
```

Comment:
Try applying this g_pub to something to see what it does..

Reply:
not sure that'd make sense (or I'm misunderstanding) - GREASEy values are only of use to check they get ignored really, which can't be done here (i do have ECH tests for that, but those'd not belong here)

Reply:
I'll mark this as resolved but do reopen if I'm missing something

Reply:
The point was that it was not a point on the curve before you changed the code :)